realsense2_camera/src/t265_realsense_node.cpp

```diff
@@ -144,3 +165,69 @@ void T265RealsenseNode::warningDiagnostic(diagnostic_updater::DiagnosticStatusWr
 {
   status.summary(diagnostic_msgs::DiagnosticStatus::WARN, _T265_fault);
 }
+
+bool T265RealsenseNode::importLocalizationMap(const std::string &localization_file)
+{
+    try
+    {
+      _pose_sensor.import_localization_map(this->bytesFromRawFile(localization_file));
+      ROS_INFO_STREAM("T265 Localization map loaded from " << localization_file);
+    }
+    catch (std::runtime_error& e)
+    {
+      ROS_WARN_STREAM("Error loading T265 map from " << localization_file << ": " << e.what());
+      return false;
+    }
+    return true;
+}
+
+bool T265RealsenseNode::exportLocalizationMap(const std::string &localization_file)
+{
+    try
+    {
+        this->rawFileFromBytes(localization_file, _pose_sensor.export_localization_map());
+        ROS_INFO_STREAM("Saved T265 map to " << localization_file);
+    }
+    catch (std::runtime_error& e)
+    {
+        ROS_WARN_STREAM("Error saving T265 map to " << localization_file << ": " << e.what());
+        return false;
+    }
+    return true;
+}
+
+bool T265RealsenseNode::saveRelocalizationMapSrv(realsense2_camera::MapPathString::Request &req,
+                                                 realsense2_camera::MapPathString::Response &res) {
+    res.success = this->exportLocalizationMap(req.filepath);
+    return true;
+}
+
+std::vector<uint8_t> T265RealsenseNode::bytesFromRawFile(const std::string &filename)

```

Comment:
this is pretty much whatever was in `bytes_from_raw_file` within `https://github.com/IntelRealSense/librealsense/blob/master/examples/ar-advanced/rs-ar-advanced.cpp`